tensorflow/c/kernels_experimental.cc

```diff
@@ -285,6 +288,97 @@ void TF_AssignUpdateVariable(TF_OpKernelContext* ctx, int input_index,
   TF_SetStatus(tf_status, TF_OK, "");
 }
 
+struct TmpVar : public ResourceBase {
+  tensorflow::mutex mu;
+  Tensor val;
+  std::string name;
+  std::string DebugString() const { return name; }
+  ~TmpVar() override { VLOG(3) << "TmpVar " << name << " deleted"; }
+};
+
+// Makes a unique name for a temporary variable inside a while loop body,
+// because loop can be executed in multiple iterations in parallel.
+std::string TemporaryVariableName(
+    const std::string& var_name,
+    const tensorflow::FrameAndIter& control_frame) {
+  if (control_frame.frame_id != tensorflow::kIllegalFrameId &&
+      control_frame.iter_id != tensorflow::kIllegalIterId) {
+    return tensorflow::strings::StrCat(var_name,
+                                       "/frame:", control_frame.frame_id,
+                                       "/iter:", control_frame.iter_id);
+  }
+  return var_name;
+}
+
+void TF_TemporaryVariable(TF_OpKernelContext* ctx, TF_DataType dtype,
+                          const int64_t* dims, int num_dims,
+                          TF_StringView* var_name,
+                          void (*allocFunc)(TF_OpKernelContext*, TF_Tensor*,
+                                            TF_DataType, const int64_t*, int,
+                                            TF_Status*),
+                          TF_Status* tf_status) {
+  auto* context = reinterpret_cast<::tensorflow::OpKernelContext*>(ctx);
+  tensorflow::ResourceMgr* rm = context->resource_manager();
+  OP_REQUIRES(context, rm,
+              tensorflow::errors::Internal("No per-step resource manager."));
```

Comment:
do you mind update to absl::Status? it is recommended to use absl::Status directly now.

Reply:
@jyingl3  done. Thanks